src/providers/GaugeProvider.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { ReactElement, useEffect, useState } from "react"
+import { useGaugeController, useHelperContract } from "../hooks/useContract"
+import { BigNumber } from "ethers"
+import { PoolTypes } from "../constants"
+
+type GaugePoolData = {
+  poolAddress: string
+  lpToken: string
+  typeOfAsset: PoolTypes
+  poolName: string
+  targetAddress: string | null
+  tokens: string[]
+  underlyingTokens: string[]
+  basePoolAddress: string
+  metaSwapDepositAddress: string
+  isSaddleApproved: boolean
+  isRemoved: boolean
+  isGuarded: boolean
+}
+
+export type Gauge = {
+  address: string
+  lpToken: string
+  name: string
+  symbol?: string
+  gaugeWeight: BigNumber
+  gaugeRelativeWeight: BigNumber
+  workingSupply?: BigNumber
+}
+
+export type Gauges = {
+  gaugeCount: number
+  gauges: Gauge[]
+} | null
+
+export const GaugeContext = React.createContext<Gauges>(null)
+
+export default function GaugeProvider({
+  children,
+}: React.PropsWithChildren<unknown>): ReactElement {
+  const gaugeController = useGaugeController()
+  const helperContract = useHelperContract()
+  const [gauges, setGauges] = useState<Gauges>(null)
+
+  useEffect(() => {
+    async function fetchGauges() {
+      const gaugeData: Gauge[] = []
+      if (!gaugeController || !helperContract) return
+      const nGauges = await gaugeController.n_gauges()
+      for (let i = 1; i <= nGauges.toNumber(); i++) {
```

Comment:
what is at index 0?

Reply:
nothing, hence it gets skipped. Also we'll be moving this to a multicall once it's ready for primetime, so this loop will go away